gtsam/inference/BayesTreeCliqueBase.h

```diff
@@ -75,10 +76,28 @@ namespace gtsam {
     /** Construct from a conditional, leaving parent and child pointers uninitialized */
     BayesTreeCliqueBase(const sharedConditional& conditional) : conditional_(conditional), problemSize_(1) {}
 
+    /** Shallow copy constructor */
```

Comment:
std::mutex has a deleted copy constructor, but BayesTreeCliqueBase implements the copy/copy assignment constructors (by default) as a shallow copy of the node in tree. To maintain the same interface as the previous BayesTreeCliqueBase, we have to implement a copy and copy assignment constructor which doesn't try and copy the cache mutex. This can still happen async because we also don't attempt to copy the cachedSeperatorMarginal, so we don't mess up the mutual exclusion of it.